crates/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -275,18 +283,29 @@ where
         &mut self,
         arg: Argument,
     ) -> Result<V, CommandArgumentError> {
+        // mutable borrowing requires unique usage
         if self.arg_is_borrowed(&arg) {
-            return Err(CommandArgumentError::InvalidUsageOfBorrowedValue);
+            return Err(CommandArgumentError::InvalidValueUsage);
         }
         self.borrowed.insert(arg, /* is_mut */ true);
         let (input_metadata_opt, val_opt) = self.borrow_mut(arg, UsageKind::BorrowMut)?;
-        if val_opt.is_none() {
-            return Err(CommandArgumentError::InvalidUsageOfTakenValue);
-        }
+        let is_copyable = if let Some(val) = val_opt {
```

Comment:
we now allow copyable values to be copied even if mutably borrowed, so we need to not take the value in that case